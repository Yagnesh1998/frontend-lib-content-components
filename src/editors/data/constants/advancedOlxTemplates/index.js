import { StrictDict } from '../../../utils';
import circuitSchematic from './circuitschematic';
import customGrader from './customgrader';
import dragAndDrop from './drag_and_drop';
import formulaResponse from './formularesponse';
import imageResponse from './imageresponse';
import jsInputResponse from './jsinput_response';
import problemWithHint from './problem_with_hint';

export default StrictDict({
  circuitSchematic, customGrader, dragAndDrop, formulaResponse, imageResponse, jsInputResponse, problemWithHint,
});
