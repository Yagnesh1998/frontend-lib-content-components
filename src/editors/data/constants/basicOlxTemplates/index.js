import { StrictDict } from '../../../utils';
import dropdown from './dropdown';
import multiSelect from './multiSelect';
import numeric from './numeric';
import singleSelect from './singleSelect';
import textInput from './textInput';

console.log(dropdown);

export default StrictDict({
  dropdown, multiSelect, numeric, singleSelect, textInput,
});
