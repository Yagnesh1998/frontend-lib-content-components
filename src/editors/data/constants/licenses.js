import { StrictDict } from '../../utils';

export const LicenseNames = StrictDict({
  allRightsReserved: 'All Rights Reserved',
  creativeCommons: 'Creative Commons',
});

export const LicenseTypes = StrictDict({
  allRightsReserved: 'all-rights-reserved',
  creativeCommons: 'creative-commons',
  // publicDomainDedication: 'public-domain-dedication', // future?
});

export const LicenseLevel = StrictDict({
  block: 'block',
  course: 'course',
  library: 'library',
})

export default {
  LicenseLevel,
  LicenseNames,
  LicenseTypes,
};
