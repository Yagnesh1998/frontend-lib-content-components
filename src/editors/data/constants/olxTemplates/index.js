import { StrictDict } from '../../../utils';

import circuitSchematic from './circuitschematic';
import customGrader from './customgrader';
import dragAndDrop from './drag_and_drop';
import formulaReponse from './formularesponse';
import imageResponse from './imageresponse';
import jsInputResponse from './jsinput_response';
import problemWithHint from './problem_with_hint';

export default StrictDict({
  circuitSchematic,
  customGrader,
  dragAndDrop,
  formulaReponse,
  imageResponse,
  jsInputResponse,
  problemWithHint,
});