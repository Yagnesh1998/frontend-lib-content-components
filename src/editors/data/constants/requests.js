import { StrictDict } from '../../utils';

export const RequestStates = StrictDict({
  inactive: 'inactive',
  pending: 'pending',
  completed: 'completed',
  failed: 'failed',
});

export const RequestKeys = StrictDict({
  fetchAssets: 'fetchAssets',
  fetchBlock: 'fetchBlock',
  fetchImages: 'fetchImages',
  fetchStudioView: 'fetchStudioView',
  fetchUnit: 'fetchUnit',
  saveBlock: 'saveBlock',
  uploadAsset: 'uploadAsset',
  allowThumbnailUpload: 'allowThumbnailUpload',
  uploadThumbnail: 'uploadThumbnail',
  uploadTranscript: 'uploadTranscript',
  deleteTranscript: 'deleteTranscript',
});
