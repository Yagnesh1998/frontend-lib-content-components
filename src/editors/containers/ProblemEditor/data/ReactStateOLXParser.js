import _ from 'lodash-es';
import { XMLParser, XMLBuilder } from 'fast-xml-parser';
import { ProblemTypeKeys } from '../../../data/constants/problem';

class ReactStateOLXParser {
  constructor(problemState) {
    const parserOptions = {
      ignoreAttributes: false,
      alwaysCreateTextNode: true,
    };
    const questionParserOptions = {
      ignoreAttributes: false,
      alwaysCreateTextNode: true,
      preserveOrder: true,
    };
    const questionBuilderOptions = {
      ignoreAttributes: false,
      attributeNamePrefix: '@_',
      suppressBooleanAttributes: false,
      format: true,
      preserveOrder: true,
    };
    const builderOptions = {
      ignoreAttributes: false,
      attributeNamePrefix: '@_',
      suppressBooleanAttributes: false,
      format: true,
    };
    this.questionParser = new XMLParser(questionParserOptions);
    this.parser = new XMLParser(parserOptions);
    this.builder = new XMLBuilder(builderOptions);
    this.questionBuilder = new XMLBuilder(questionBuilderOptions);
    this.editorObject = problemState.editorObject;
    this.problemState = problemState.problem;
  }

  addHints() {
    const hintsArray = [];
    const hints = _.get(this.problemState, 'settings.hints', []);
    hints.forEach(element => {
      hintsArray.push({
        '#text': element.value,
      });
    });
    const demandhint = {
      demandhint: {
        hint: hintsArray,
      },
    };
    return demandhint;
  }

  addSolution() {
    if (!_.has(this.problemState, 'settings.solutionExplanation')) { return {}; }
    const { solution } = this.editorObject;
    const parsedSolution = this.parser.parse(solution);
    const solutionObject = {
      solution: {
        ...parsedSolution,
      },
    };
    return solutionObject;
  }

  addMultiSelectAnswers(option) {
    const choice = [];
    let compoundhint = [];
    let widget = {};
    // eslint-disable-next-line prefer-const
    let { answers, generalFeedback } = this.problemState;
    const answerTitles = this.editorObject?.answers;
    // general feedback replaces selected feedback if all incorrect selected feedback is the same.
    if (generalFeedback !== ''
    && answers.every(
      answer => (
        answer.correct
          ? true
          : answer?.selectedFeedback === answers.find(a => a.correct === false).selectedFeedback
      ),
    )) {
      answers = answers.map(answer => (!answer?.correct
        ? { ...answer, selectedFeedback: generalFeedback }
        : answer));
    }
    answers.forEach((answer) => {
      const feedback = [];
      let singleAnswer = {};
      const title = answerTitles ? answerTitles[answer.id] : answer.title;
      if (this.hasAttributeWithValue(answer, 'title')) {
        if (this.hasAttributeWithValue(answer, 'selectedFeedback')) {
          feedback.push({
            '#text': _.get(answer, 'selectedFeedback'),
            '@_selected': true,
          });
        }
        if (this.hasAttributeWithValue(answer, 'unselectedFeedback')) {
          feedback.push({
            '#text': _.get(answer, 'unselectedFeedback'),
            '@_selected': false,
          });
        }
        if (this.hasAttributeWithValue(answer, 'feedback')) {
          feedback.push({
            '#text': _.get(answer, 'feedback'),
          });
        }
        if (feedback.length) {
          singleAnswer[`${option}hint`] = feedback;
        }
        singleAnswer = {
          '#text': title,
          '@_correct': answer.correct,
          ...singleAnswer,
        };
        choice.push(singleAnswer);
      }
    });
    widget = { [option]: choice };
    if (_.has(this.problemState, 'groupFeedbackList')) {
      compoundhint = this.addGroupFeedbackList();
      widget = {
        ...widget,
        compoundhint,
      };
    }
    return widget;
  }

  addGroupFeedbackList() {
    const compoundhint = [];
    const { groupFeedbackList } = this.problemState;
    groupFeedbackList.forEach((element) => {
      compoundhint.push({
        '#text': element.feedback,
        '@_value': element.answers.join(' '),
      });
    });
    return compoundhint;
  }

  addQuestion() {
    const question = this.editorObject?.question || this.problemState?.question;
    // const question = this.editorObject?.question || '';
    const questionObject = this.questionParser.parse(question);
    return questionObject;
  }

  buildMultiSelectProblem(problemType, widget, option) {
    const question = this.addQuestion();
    const widgetObject = this.addMultiSelectAnswers(option);
    const demandhint = this.addHints();
    const solution = this.addSolution();

    const problemObject = {
      problem: {
        [problemType]: {
          [widget]: widgetObject,
          ...solution,
        },
        ...demandhint,
      },
    };

    const problem = this.builder.build(problemObject);
    const questionString = this.questionBuilder.build(question);
    let problemTypeTag;
    switch (problemType) {
      case ProblemTypeKeys.MULTISELECT:
        [problemTypeTag] = problem.match(/<choiceresponse>|<choiceresponse.[^>]+>/);
        break;
      case ProblemTypeKeys.DROPDOWN:
        [problemTypeTag] = problem.match(/<optionresponse>|<optionresponse.[^>]+>/);
        break;
      case ProblemTypeKeys.SINGLESELECT:
        [problemTypeTag] = problem.match(/<multiplechoiceresponse>|<multiplechoiceresponse.[^>]+>/);
        break;
      default:
        break;
    }
    const updatedString = `${problemTypeTag}\n${questionString}`;
    const problemString = problem.replace(problemTypeTag, updatedString);

    return problemString;
  }

  buildTextInput() {
    const question = this.addQuestion();
    const demandhint = this.addHints();
    const answerObject = this.buildTextInputAnswersFeedback();
    const solution = this.addSolution();

    const problemObject = {
      problem: {
        [ProblemTypeKeys.TEXTINPUT]: {
          ...answerObject,
          ...solution,
        },
        ...demandhint,
      },
    };

    const problem = this.builder.build(problemObject);
    const questionString = this.questionBuilder.build(question);
    const [problemTypeTag] = problem.match(/<stringresponse>|<stringresponse.[^>]+>/);
    const updatedString = `${problemTypeTag}\n${questionString}`;
    const problemString = problem.replace(problemTypeTag, updatedString);

    return problemString;
  }

  buildTextInputAnswersFeedback() {
    const { answers } = this.problemState;
    const answerTitles = this.editorObject?.answers;
    let answerObject = {};
    const additionAnswers = [];
    const wrongAnswers = [];
    let firstCorrectAnswerParsed = false;
    answers.forEach((answer) => {
      const correcthint = this.getAnswerHints(answer);
      if (this.hasAttributeWithValue(answer, 'title')) {
        const title = answerTitles ? answerTitles[answer.id] : answer.title;
        if (answer.correct && firstCorrectAnswerParsed) {
          additionAnswers.push({
            '@_answer': title,
            ...correcthint,
          });
        } else if (answer.correct && !firstCorrectAnswerParsed) {
          firstCorrectAnswerParsed = true;
          answerObject = {
            '@_answer': title,
            ...correcthint,
          };
        } else if (!answer.correct) {
          wrongAnswers.push({
            '@_answer': title,
            '#text': answer.selectedFeedback,
          });
        }
      }
    });
    answerObject = {
      ...answerObject,
      additional_answer: additionAnswers,
      stringequalhint: wrongAnswers,
      '@_type': _.get(this.problemState, 'additionalAttributes.type', 'ci'),
      textline: {
        '@_size': _.get(this.problemState, 'additionalAttributes.textline.size', 20),
      },
    };
    return answerObject;
  }

  buildNumericInput() {
    const question = this.addQuestion();
    const demandhint = this.addHints();
    const answerObject = this.buildNumericalResponse();
    const solution = this.addSolution();

    const problemObject = {
      problem: {
        [ProblemTypeKeys.NUMERIC]: {
          ...answerObject,
          ...solution,
        },
        ...demandhint,
      },
    };

    const problem = this.builder.build(problemObject);
    const questionString = this.questionBuilder.build(question);
    const [problemTypeTag] = problem.match(/<numericalresponse>|<numericalresponse.[^>]+>/);
    const updatedString = `${questionString}\n${problemTypeTag}`;
    const problemString = problem.replace(problemTypeTag, updatedString);

    return problemString;
  }

  buildNumericalResponse() {
    const { answers } = this.problemState;
    const answerTitles = this.editorObject?.answers;
    let answerObject = {};
    const additionalAnswers = [];
    let firstCorrectAnswerParsed = false;
    answers.forEach((answer) => {
      const correcthint = this.getAnswerHints(answer);
      if (this.hasAttributeWithValue(answer, 'title')) {
        const title = answerTitles ? answerTitles[answer.id] : answer.title;
        if (answer.correct && !firstCorrectAnswerParsed) {
          firstCorrectAnswerParsed = true;
          let responseParam = {};
          if (_.has(answer, 'tolerance')) {
            responseParam = {
              responseparam: {
                '@_type': 'tolerance',
                '@_default': _.get(answer, 'tolerance', 0),
              },
            };
          }
          answerObject = {
            '@_answer': title,
            ...responseParam,
            ...correcthint,
          };
        } else if (answer.correct && firstCorrectAnswerParsed) {
          additionalAnswers.push({
            '@_answer': title,
            ...correcthint,
          });
        }
      }
    });
    answerObject = {
      ...answerObject,
      additional_answer: additionalAnswers,
      formulaequationinput: {
        '#text': '',
      },
    };
    return answerObject;
  }

  getAnswerHints(elementObject) {
    const feedback = elementObject?.selectedFeedback;
    let correcthint = {};
    if (feedback !== undefined && feedback !== '') {
      correcthint = {
        correcthint: {
          '#text': feedback,
        },
      };
    }
    return correcthint;
  }

  hasAttributeWithValue(obj, attr) {
    return _.has(obj, attr) && _.get(obj, attr, '').toString().trim() !== '';
  }

  buildOLX() {
    const { problemType } = this.problemState;
    let problemString = '';

    switch (problemType) {
      case ProblemTypeKeys.MULTISELECT:
        problemString = this.buildMultiSelectProblem(ProblemTypeKeys.MULTISELECT, 'checkboxgroup', 'choice');
        break;
      case ProblemTypeKeys.DROPDOWN:
        problemString = this.buildMultiSelectProblem(ProblemTypeKeys.DROPDOWN, 'optioninput', 'option');
        break;
      case ProblemTypeKeys.SINGLESELECT:
        problemString = this.buildMultiSelectProblem(ProblemTypeKeys.SINGLESELECT, 'choicegroup', 'choice');
        break;
      case ProblemTypeKeys.TEXTINPUT:
        problemString = this.buildTextInput();
        break;
      case ProblemTypeKeys.NUMERIC:
        problemString = this.buildNumericInput();
        break;
      default:
        break;
    }
    return problemString;
  }
}

export default ReactStateOLXParser;
