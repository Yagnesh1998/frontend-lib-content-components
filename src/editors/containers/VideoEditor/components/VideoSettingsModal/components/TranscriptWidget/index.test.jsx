import React from 'react';
import { shallow } from 'enzyme';

import { formatMessage } from '../../../../../../../testUtils';
import { actions, selectors } from '../../../../../../data/redux';
import { RequestKeys } from '../../../../../../data/constants/requests';

import { TranscriptWidget, mapStateToProps, mapDispatchToProps } from '.';

jest.mock('../../../../../../data/redux', () => ({
  actions: {
    video: {
      updateField: jest.fn().mockName('actions.video.updateField'),
    },
  },
  selectors: {
    video: {
      transcripts: jest.fn(state => ({ transcripts: state })),
      allowTranscriptDownloads: jest.fn(state => ({ allowTranscriptDownloads: state })),
      showTranscriptByDefault: jest.fn(state => ({ showTranscriptByDefault: state })),
    },
    requests: {
      isFailed: jest.fn(state => ({ isFailed: state })),
    },
  },
}));

describe('TranscriptWidget', () => {
  const props = {
    error: {},
    subtitle: 'SuBTItle',
    title: 'tiTLE',
    // inject
    intl: { formatMessage },
    // redux
    transcripts: null,
    allowTranscriptDownloads: false,
    showTranscriptByDefault: false,
    updateField: jest.fn().mockName('args.updateField'),
    isUploadError: false,
    isDeleteError: false,
  };

  describe('snapshots', () => {
    test('snapshots: renders as expected with default props', () => {
      expect(
        shallow(<TranscriptWidget {...props} />),
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with transcripts', () => {
      expect(
        shallow(<TranscriptWidget {...props} transcripts={{ english: 'sOMeUrl' }} />),
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with allowTranscriptDownloads true', () => {
      expect(
        shallow(<TranscriptWidget {...props} allowTranscriptDownloads transcripts={{ english: 'sOMeUrl' }} />),
      ).toMatchSnapshot();
    });
    test('snapshots: renders as expected with showTranscriptByDefault true', () => {
      expect(
        shallow(<TranscriptWidget {...props} showTranscriptByDefault transcripts={{ english: 'sOMeUrl' }} />),
      ).toMatchSnapshot();
    });
    test('snapshot: renders ErrorAlert with upload error message', () => {
      expect(
        shallow(<TranscriptWidget {...props} isUploadError transcripts={{ english: 'sOMeUrl' }} />),
      ).toMatchSnapshot();
    });
    test('snapshot: renders ErrorAlert with delete error message', () => {
      expect(
        shallow(<TranscriptWidget {...props} isDeleteError transcripts={{ english: 'sOMeUrl' }} />),
      ).toMatchSnapshot();
    });
  });
  describe('mapStateToProps', () => {
    const testState = { A: 'pple', B: 'anana', C: 'ucumber' };
    test('transcripts from video.transcript', () => {
      expect(
        mapStateToProps(testState).transcripts,
      ).toEqual(selectors.video.transcripts(testState));
    });
    test('allowTranscriptDownloads from video.allowTranscriptDownloads', () => {
      expect(
        mapStateToProps(testState).allowTranscriptDownloads,
      ).toEqual(selectors.video.allowTranscriptDownloads(testState));
    });
    test('showTranscriptByDefault from video.showTranscriptByDefault', () => {
      expect(
        mapStateToProps(testState).showTranscriptByDefault,
      ).toEqual(selectors.video.showTranscriptByDefault(testState));
    });
    test('showTranscriptByDefault from video.showTranscriptByDefault', () => {
      expect(
        mapStateToProps(testState).showTranscriptByDefault,
      ).toEqual(selectors.video.showTranscriptByDefault(testState));
    });
    test('isUploadError from requests.isFinished', () => {
      expect(
        mapStateToProps(testState).isUploadError,
      ).toEqual(selectors.requests.isFailed(testState, { requestKey: RequestKeys.uploadTranscript }));
    });
    test('isDeleteError from requests.isFinished', () => {
      expect(
        mapStateToProps(testState).isDeleteError,
      ).toEqual(selectors.requests.isFailed(testState, { requestKey: RequestKeys.deleteTranscript }));
    });
  });
  describe('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    test('updateField from actions.video.updateField', () => {
      expect(mapDispatchToProps.updateField).toEqual(dispatch(actions.video.updateField));
    });
  });
});
