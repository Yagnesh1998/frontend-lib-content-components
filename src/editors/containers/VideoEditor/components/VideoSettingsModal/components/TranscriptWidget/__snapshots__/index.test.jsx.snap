// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranscriptWidget snapshots snapshot: renders ErrorAlert with delete error message 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="English"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={true}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="mt-4.5"
    >
      <injectIntl(ShimmedIntlComponent)
        language="en"
      />
      <div
        className="mb-1"
      >
        <Form.Checkbox
          checked={false}
          className="mt-4.5 decorative-control-label"
          onChange={[Function]}
        >
          <Form.Label>
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </Form.Label>
        </Form.Checkbox>
        <OverLayTrigger
          key="right"
          overlay={
            <ToolTip>
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </ToolTip>
          }
          placement="right"
        >
          <Icon
            className="d-inline-block mx-3"
          />
        </OverLayTrigger>
      </div>
      <Form.Checkbox
        checked={false}
        className="mt-4.5 decorative-control-label"
        onChange={[Function]}
      >
        <Form.Label
          size="sm"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </Form.Label>
      </Form.Checkbox>
    </Form.Group>
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;

exports[`TranscriptWidget snapshots snapshot: renders ErrorAlert with upload error message 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="English"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={true}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="mt-4.5"
    >
      <injectIntl(ShimmedIntlComponent)
        language="en"
      />
      <div
        className="mb-1"
      >
        <Form.Checkbox
          checked={false}
          className="mt-4.5 decorative-control-label"
          onChange={[Function]}
        >
          <Form.Label>
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </Form.Label>
        </Form.Checkbox>
        <OverLayTrigger
          key="right"
          overlay={
            <ToolTip>
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </ToolTip>
          }
          placement="right"
        >
          <Icon
            className="d-inline-block mx-3"
          />
        </OverLayTrigger>
      </div>
      <Form.Checkbox
        checked={false}
        className="mt-4.5 decorative-control-label"
        onChange={[Function]}
      >
        <Form.Label
          size="sm"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </Form.Label>
      </Form.Checkbox>
    </Form.Group>
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;

exports[`TranscriptWidget snapshots snapshots: renders as expected with allowTranscriptDownloads true 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="English"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="mt-4.5"
    >
      <injectIntl(ShimmedIntlComponent)
        language="en"
      />
      <div
        className="mb-1"
      >
        <Form.Checkbox
          checked={true}
          className="mt-4.5 decorative-control-label"
          onChange={[Function]}
        >
          <Form.Label>
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </Form.Label>
        </Form.Checkbox>
        <OverLayTrigger
          key="right"
          overlay={
            <ToolTip>
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </ToolTip>
          }
          placement="right"
        >
          <Icon
            className="d-inline-block mx-3"
          />
        </OverLayTrigger>
      </div>
      <Form.Checkbox
        checked={false}
        className="mt-4.5 decorative-control-label"
        onChange={[Function]}
      >
        <Form.Label
          size="sm"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </Form.Label>
      </Form.Checkbox>
    </Form.Group>
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;

exports[`TranscriptWidget snapshots snapshots: renders as expected with default props 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="None"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Alert
      variant="danger"
    >
      <FormattedMessage
        defaultMessage="Only SRT files can be uploaded. Please select a file ending in .srt to upload."
        description="Message warning users to only upload .srt files"
        id="authoring.videoeditor.transcripts.fileTypeWarning"
      />
    </Alert>
    <FormattedMessage
      defaultMessage="Add video transcripts for improved accessibility."
      description="Message for adding first transcript"
      id="authoring.videoeditor.transcripts.upload.firstTranscriptMessage"
    />
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;

exports[`TranscriptWidget snapshots snapshots: renders as expected with showTranscriptByDefault true 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="English"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="mt-4.5"
    >
      <injectIntl(ShimmedIntlComponent)
        language="en"
      />
      <div
        className="mb-1"
      >
        <Form.Checkbox
          checked={false}
          className="mt-4.5 decorative-control-label"
          onChange={[Function]}
        >
          <Form.Label>
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </Form.Label>
        </Form.Checkbox>
        <OverLayTrigger
          key="right"
          overlay={
            <ToolTip>
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </ToolTip>
          }
          placement="right"
        >
          <Icon
            className="d-inline-block mx-3"
          />
        </OverLayTrigger>
      </div>
      <Form.Checkbox
        checked={true}
        className="mt-4.5 decorative-control-label"
        onChange={[Function]}
      >
        <Form.Label
          size="sm"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </Form.Label>
      </Form.Checkbox>
    </Form.Group>
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;

exports[`TranscriptWidget snapshots snapshots: renders as expected with transcripts 1`] = `
<injectIntl(ShimmedIntlComponent)
  isError={true}
  subtitle="English"
  title="Transcript"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="mt-4.5"
    >
      <injectIntl(ShimmedIntlComponent)
        language="en"
      />
      <div
        className="mb-1"
      >
        <Form.Checkbox
          checked={false}
          className="mt-4.5 decorative-control-label"
          onChange={[Function]}
        >
          <Form.Label>
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </Form.Label>
        </Form.Checkbox>
        <OverLayTrigger
          key="right"
          overlay={
            <ToolTip>
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </ToolTip>
          }
          placement="right"
        >
          <Icon
            className="d-inline-block mx-3"
          />
        </OverLayTrigger>
      </div>
      <Form.Checkbox
        checked={false}
        className="mt-4.5 decorative-control-label"
        onChange={[Function]}
      >
        <Form.Label
          size="sm"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </Form.Label>
      </Form.Checkbox>
    </Form.Group>
    <FileInput
      acceptedFiles=".srt"
      fileInput={
        Object {
          "addFile": [Function],
          "click": [Function],
          "ref": Object {
            "current": undefined,
          },
        }
      }
    />
    <Button
      onClick={[Function]}
      variant="link"
    >
      <FormattedMessage
        defaultMessage="Upload Transcript"
        description="Label for upload button"
        id="authoring.videoeditor.transcripts.upload.label"
      />
    </Button>
  </Stack>
</injectIntl(ShimmedIntlComponent)>
`;
